server/msteams/client_disconnectionlayer/disconnectionlayer.go

```diff
@@ -0,0 +1,622 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```

Comment:
This is the generated layer for the disconnection detection. The main point is that you can create the layer on top of the client and if the returned error is a 401 error we call the onDisconnect callback with the user id provided in the constructor.